Description: add __FreeBSD_kernel__ to the list of *bsd conditions
Forwarded: not-needed, not yet tested enough
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2015-01-21
Index: nodejs/deps/uv/include/uv-unix.h
===================================================================
--- nodejs.orig/deps/uv/include/uv-unix.h
+++ nodejs/deps/uv/include/uv-unix.h
@@ -50,6 +50,7 @@
 # include "uv-darwin.h"
 #elif defined(__DragonFly__)  || \
       defined(__FreeBSD__)    || \
+      defined(__FreeBSD_kernel__) || \
       defined(__OpenBSD__)    || \
       defined(__NetBSD__)
 # include "uv-bsd.h"
Index: nodejs/deps/uv/src/unix/core.c
===================================================================
--- nodejs.orig/deps/uv/src/unix/core.c
+++ nodejs/deps/uv/src/unix/core.c
@@ -54,7 +54,7 @@
 # include <sys/ioctl.h>
 #endif
 
-#ifdef __FreeBSD__
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 # include <sys/sysctl.h>
 # include <sys/filio.h>
 # include <sys/ioctl.h>
@@ -444,7 +444,7 @@ int uv__close(int fd) {
 }
 
 
-#if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__APPLE__)
 
 int uv__nonblock(int fd, int set) {
   int r;
@@ -473,7 +473,7 @@ int uv__cloexec(int fd, int set) {
   return 0;
 }
 
-#else /* !(defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)) */
+#else /* !(defined(__linux__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__APPLE__)) */
 
 int uv__nonblock(int fd, int set) {
   int flags;
@@ -536,7 +536,7 @@ int uv__cloexec(int fd, int set) {
   return 0;
 }
 
-#endif /* defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__) */
+#endif /* defined(__linux__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__APPLE__) */
 
 
 /* This function is not execve-safe, there is a race window
Index: nodejs/deps/uv/src/unix/fs.c
===================================================================
--- nodejs.orig/deps/uv/src/unix/fs.c
+++ nodejs/deps/uv/src/unix/fs.c
@@ -49,7 +49,7 @@
     defined(__OpenBSD__)    ||                                            \
     defined(__NetBSD__)
 # define HAVE_PREADV 1
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__FreeBSD_kernel__)
 # include <linux/version.h>
 # if defined(__GLIBC_PREREQ)
 #   if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,30) &&                    \
@@ -194,6 +194,7 @@ skip:
 #elif defined(__APPLE__)                                                      \
     || defined(__DragonFly__)                                                 \
     || defined(__FreeBSD__)                                                   \
+    || defined(__FreeBSD_kernel__)                                            \
     || defined(__NetBSD__)                                                    \
     || defined(__OpenBSD__)                                                   \
     || defined(__sun)
Index: nodejs/deps/uv/test/test-embed.c
===================================================================
--- nodejs.orig/deps/uv/test/test-embed.c
+++ nodejs/deps/uv/test/test-embed.c
@@ -29,6 +29,7 @@
 # if defined(__APPLE__) ||                                                    \
      defined(__DragonFly__) ||                                                \
      defined(__FreeBSD__) ||                                                  \
+     defined(__FreeBSD_kernel__) ||                                           \
      defined(__OpenBSD__) ||                                                  \
      defined(__NetBSD__)
 #  define HAVE_KQUEUE 1
Index: nodejs/deps/uv/test/test-fs-event.c
===================================================================
--- nodejs.orig/deps/uv/test/test-fs-event.c
+++ nodejs/deps/uv/test/test-fs-event.c
@@ -29,6 +29,7 @@
 # if defined(__APPLE__) ||                                                    \
      defined(__DragonFly__) ||                                                \
      defined(__FreeBSD__) ||                                                  \
+     defined(__FreeBSD_kernel__) ||                                           \
      defined(__OpenBSD__) ||                                                  \
      defined(__NetBSD__)
 #  define HAVE_KQUEUE 1
Index: nodejs/tools/install.py
===================================================================
--- nodejs.orig/tools/install.py
+++ nodejs/tools/install.py
@@ -135,7 +135,7 @@ def files(action):
   # behave similarly for systemtap
   action(['src/node.stp'], 'share/systemtap/tapset/')
 
-  if 'freebsd' in sys.platform or 'openbsd' in sys.platform:
+  if ('freebsd' in sys.platform and not 'kfreebsd' in sys.platform)  or 'openbsd' in sys.platform:
     action(['doc/node.1'], 'man/man1/')
   else:
     action(['doc/node.1'], 'share/man/man1/')
Index: nodejs/deps/uv/common.gypi
===================================================================
--- nodejs.orig/deps/uv/common.gypi
+++ nodejs/deps/uv/common.gypi
@@ -130,7 +130,7 @@
           }]
         ]
       }],
-      ['OS in "freebsd linux openbsd solaris android"', {
+      ['OS in "freebsd linux openbsd solaris android kfreebsd"', {
         'cflags': [ '-Wall' ],
         'cflags_cc': [ '-fno-rtti', '-fno-exceptions' ],
         'target_conditions': [
Index: nodejs/deps/uv/uv.gyp
===================================================================
--- nodejs.orig/deps/uv/uv.gyp
+++ nodejs/deps/uv/uv.gyp
@@ -250,7 +250,7 @@
             ],
           },
         }],
-        [ 'OS=="freebsd" or OS=="dragonflybsd"', {
+        [ 'OS=="freebsd" or OS=="dragonflybsd" or OS=="kfreebsd"', {
           'sources': [ 'src/unix/freebsd.c' ],
         }],
         [ 'OS=="openbsd"', {
@@ -264,7 +264,12 @@
             'libraries': [ '-lkvm' ],
           },
         }],
-        [ 'OS in "mac freebsd dragonflybsd openbsd netbsd".split()', {
+	[ 'OS=="kfreebsd"', {
+          'link_settings': {
+            'libraries': [ '-lkvm -ldl' ],
+          },
+        }],
+        [ 'OS in "mac freebsd kfreebsd dragonflybsd openbsd netbsd".split()', {
           'sources': [ 'src/unix/kqueue.c' ],
         }],
         ['uv_library=="shared_library"', {
Index: nodejs/node.gyp
===================================================================
--- nodejs.orig/node.gyp
+++ nodejs/node.gyp
@@ -315,6 +315,11 @@
             '-lkvm',
           ],
         }],
+	[ 'OS=="kfreebsd"', {
+          'libraries': [
+            '-lkvm',
+          ],
+        }],
         [ 'OS=="solaris"', {
           'libraries': [
             '-lkstat',
Index: nodejs/common.gypi
===================================================================
--- nodejs.orig/common.gypi
+++ nodejs/common.gypi
@@ -162,7 +162,7 @@
         'cflags': [ '-pthread', ],
         'ldflags': [ '-pthread' ],
       }],
-      [ 'OS in "linux freebsd openbsd solaris android"', {
+      [ 'OS in "linux freebsd kfreebsd openbsd solaris android"', {
         'cflags': [ '-Wall', '-Wextra', '-Wno-unused-parameter', ],
         'cflags_cc': [ '-fno-rtti', '-fno-exceptions' ],
         'ldflags': [ '-rdynamic' ],
@@ -230,7 +230,7 @@
           }],
         ],
       }],
-      ['OS=="freebsd" and node_use_dtrace=="true"', {
+      ['(OS=="freebsd" or OS=="kfreebsd") and node_use_dtrace=="true"', {
         'libraries': [ '-lelf' ],
       }]
     ],
